### PR TITLE
refactor(core): use utxolib.bitgo.verifySignature()

### PR DESCRIPTION
**Pull request by OttoAllmendinger**
Use functionally identical `verifySignature()` method from utxolib.

Issue: BG-34381

### CODE REVIEW

File: modules/core/test/v2/unit/coins/btc.ts [hunk 1]
```diff
@@ -320,20 +321,20 @@ describe('BTC:', function () {
         },
       };
 
-      let coin;
+      let coin: AbstractUtxoCoin;
```
Comment (sangaman) [inline modules/core/test/v2/unit/coins/btc.ts:324]: I'm curious why you specified a type here if you have to cast to `any` below?

**Review by sangaman — APPROVED** [review]
Nice deduplification.

File: modules/core/test/v2/unit/coins/btc.ts (hunk 1 above)
Comment (OttoAllmendinger) [reply modules/core/test/v2/unit/coins/btc.ts:324]: to get IntelliJ support :)

Comment (OttoAllmendinger) [reply modules/core/test/v2/unit/coins/btc.ts:324]: the tests where I have to cast to `any` are simulating use from untyped JS

**Pull request merged**